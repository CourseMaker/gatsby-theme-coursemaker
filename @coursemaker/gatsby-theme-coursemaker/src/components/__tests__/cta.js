import React from 'react';
import renderer from 'react-test-renderer';
import CTA from '../cta';
import priceInfoDouble from '../__mocks__/course-price-double';

describe('CTA', () => {
    it('renders correctly for one-time purchase', () => {
        const priceInfoSample = priceInfoDouble;
        const tree = renderer.create(<CTA priceInfo={priceInfoSample} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('CTA', () => {
    it('renders correctly for membership purchase', () => {
        const priceInfoSample = priceInfoDouble;
        priceInfoSample.product_type = 'school_membership';
        priceInfoSample.recurring_interval = 'month';
        const tree = renderer.create(<CTA priceInfo={priceInfoSample} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
