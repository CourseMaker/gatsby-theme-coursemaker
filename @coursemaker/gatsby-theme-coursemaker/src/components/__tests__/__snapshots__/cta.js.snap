// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA renders correctly for membership purchase 1`] = `
<div
  className="mt-8 btn-wrapper"
>
  <a
    className="btn btn-lg css-1v6n45o-Button"
    href="./checkout"
    text="Purchase Membership2"
  >
    Purchase Membership2
  </a>
</div>
`;

exports[`CTA renders correctly for one-time purchase 1`] = `
<div
  className="mt-8 btn-wrapper"
>
  <a
    className="btn btn-lg css-1v6n45o-Button"
    href="./checkout"
    text="Purchase Course"
  >
    Purchase Course
  </a>
</div>
`;
